api/proto/config/storage.proto

```diff
@@ -242,3 +243,48 @@ message DisksConfig {
   DisksArrayType array_type = 2;
   repeated DisksConfig children = 3;
 }
+
+// SnapshotOpsCmd counter for snapshot cmd
+message SnapshotOpsCmd {
+  uint32 counter = 1;
+  google.protobuf.Timestamp opsTime = 2;
+}
+
+// ZfsSnapshot describes a snapshot for a specific logical
+// volume that must exist on the device. It has a required
+// field volume_uuid for which it was created.
+message ZfsSnapshot {
+  // The real snapshot name in ZFS
+  string id = 1;
+
+  // Display name (User-friendly name). This name does not affect
+  // the snapshot properties in ZFS in any way
+  string display_name = 2;
+
+  // ID of the volume this snapshot belongs to. The field is required
+  // for all commands except for the rename command.
+  string volume_uuid = 3;
+
+  // Time of the last operation on the snapshot
+  uint64 last_cmd_timestamp = 4;
+
+  // Flag indicating that the snapshot has been deleted
+  bool deleted = 5;
+
+  // The command for creating a snapshot, for execution, requires the presence
+  // of a volume_uuid for which the snapshot will be created. By default, populates
+  // display_name with the date the snapshot was taken.
+  SnapshotOpsCmd create = 6;
+
+  // The command to delete a snapshot in ZFS.
+  SnapshotOpsCmd destroy = 7;
+
+  // The command to change the value for display_name.
+  SnapshotOpsCmd rename = 8;
```

Comment:
@giggsoff 
Initially, I thought that when creating a snapshot, you need to give the snapshot a name in the form of the date it was created in order to send this information to the cloud, in which case the snapshot will have some kind of friendly name. In this case, the rename command seems logical.

Your comment makes sense since on the EVE side we don't care what the name of the snapshot is somewhere in the cloud. I don't know how it will be implemented on the cloud side, but if we can avoid storing the user-friendly name on the EVE side, and only store it in the cloud. then it will be good. 

In this case, this command will work and exist only in the cloud, and when creating a snapshot, the name will be given by the cloud. But for some reason, it seems to me that when creating a snapshot, the cloud will not know what ID the snapshot will have, and how it will need to be associated with a user-friendly name.